File: language-samples/hello.m
/**
 * Objective-C sample for GitHub language detection
 * BarbrickDesign - Complete Language Portfolio
 */

#import <Foundation/Foundation.h>

@interface Greeter : NSObject
@property (nonatomic, strong) NSString *message;
- (instancetype)initWithMessage:(NSString *)message;
- (void)greet;
@end

@implementation Greeter

- (instancetype)initWithMessage:(NSString *)message {
    self = [super init];
    if (self) {
        _message = message;
    }
    return self;
}

- (void)greet {
    NSLog(@"%@", self.message);
}

@end

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        Greeter *greeter = [[Greeter alloc] initWithMessage:@"Hello from Objective-C!"];
        [greeter greet];
        
        // Array
        NSArray *numbers = @[@1, @2, @3, @4, @5];
        NSMutableArray *doubled = [NSMutableArray array];
        for (NSNumber *num in numbers) {
            [doubled addObject:@([num intValue] * 2)];
        }
        
        // Dictionary
        NSDictionary *info = @{
            @"language": @"Objective-C",
            @"paradigm": @"Object-oriented",
            @"typing": @"Static"
        };
        
        NSLog(@"%@", info);
    }
    return 0;
}

File: language-samples/hello.swift
/**
 * Swift sample for GitHub language detection
 * BarbrickDesign - Complete Language Portfolio
 */

import Foundation

class Greeter {
    var message: String
    
    init(message: String) {
        self.message = message
    }
    
    func greet() {
        print(message)
    }
}

func main() {
    let greeter = Greeter(message: "Hello from Swift!")
    greeter.greet()
    
    // Array
    let numbers = [1, 2, 3, 4, 5]
    let doubled = numbers.map { $0 * 2 }
    
    // Dictionary
    let info: [String: String] = [
        "language": "Swift",
        "paradigm": "Multi-paradigm",
        "typing": "Static"
    ]
    
    // For-in loop
    let languages = ["Swift", "Objective-C", "C"]
    for lang in languages {
        print(lang.uppercased())
    }
    
    // Optional
    var optional: String? = "Optional value"
    if let value = optional {
        print(value)
    }
}

main()
